flow360/component/resource_base.py

```diff
@@ -377,3 +395,184 @@ def from_cloud(cls):
         get ResourceList from cloud
         """
         return cls(from_cloud=True)
+
+
+class RemoteResourceLogs:
+    """
+    Logs class for getting remote logs from flow360 resources
+    """
+
+    def __init__(self, flow360: Flow360Resource):
+        self.dir_path = None
+        self.path = None
+        self.paths = []
+        self.flow360 = flow360
+
+    def download_log(
+        self,
+        to_file="./logs/",
+    ):
+        """
+        Download log file from surface mesh.
+
+        :param to_file: Destination folder path to save the log file(s).
+        :param keep_folder: Flag to indicate whether to keep the folder structure while downloading.
+        :param overwrite: Flag to indicate whether to overwrite existing files.
+        :param progress_callback: Optional callback function for progress updates.
+        :param kwargs: Additional keyword arguments passed to Flow360Resource.download_file().
+        """
+        self.dir_path = to_file
+        file_list = self.flow360.get_download_file_list()
+        regex = re.compile(r"logs/.*\.log")
+        for fname in file_list:
+            if regex.match(fname):
+                self.flow360.download_file(fname, to_file)
```

Comment:
will it download every time I want access? I think we might need a function 'update' or 'refresh' to parse latest logs. Generally, when resource is in final status (`self.status.is_final()`) the logs should not change anymore. Can you check with Lei if resource status is set to final after uploading is done or before? I think it is after.